packages/contractkit/src/identity/offchain-data-wrapper.ts

```diff
@@ -52,35 +69,34 @@ class StorageRoot {
   constructor(readonly account: string, readonly address: string) {}
 
   // TODO: Add decryption metadata (i.e. indicates ciphertext to be decrypted/which key to use)
-  async read(dataPath: string): Promise<[any, any]> {
+  async read(dataPath: string): Promise<string> {
     const data = await fetch(this.address + dataPath)
     if (!data.ok) {
-      return [null, 'No can do']
+      throw new FetchError()
     }
 
     const body = await data.text()
 
     const signature = await fetch(this.address + dataPath + '.signature')
 
     if (!signature.ok) {
-      return [null, 'Signature could not be fetched']
+      throw new FetchError()
     }
 
     // TODO: Compare against registered on-chain signers or off-chain signers
     const signer = guessSigner(body, await signature.text())
 
     if (eqAddress(signer, this.account)) {
-      return [body, null]
+      return body
     }
 
     // The signer might be authorized off-chain
     // TODO: Only if the signer is authorized with an on-chain key
-    const [, err] = await this.read(`/account/authorizedSigners/${toChecksumAddress(signer)}`)
-
-    if (err) {
-      return [null, err]
+    try {
+      await this.read(`/account/authorizedSigners/${toChecksumAddress(signer)}`)
+      return body
+    } catch (error) {
+      throw error
```

Comment:
How do 'unchecked' exception (to borrow a Java term) fit in to the whole scheme here?